src/Reducer.js

```diff
@@ -246,6 +246,47 @@ function inject(state, action, props, scenes) {
         from: null,
         children: state.children,
       };
+    case ActionConst.MODIFY_STACK: {
+      const newChildren = [...state.children];
+      let removedIndex = null;
+
+      assert(props.commands, 'Modify stack commands is undefined.');
+      props.commands.forEach(command => {
+        const findIndex = sceneKey => {
+          const index = newChildren.findIndex(c => c.sceneKey === sceneKey);
+          assert(index !== -1,
+            `The sceneKey: ${sceneKey} could not be found in the stack - ${command}`);
+          return index;
+        };
+        switch (command.type) {
+          case ActionConst.ModifyStackTypes.REMOVE:
+            assert(command.sceneKey || command.index >= 0,
+              `sceneKey or index has to be defined - ${command}`);
+            removedIndex = command.sceneKey ? findIndex(command.sceneKey) : command.index;
+            assert(removedIndex < newChildren.length - 1,
+              `You are not allowed to remove current scene - ${command}`);
+            newChildren.splice(removedIndex, 1);
+            break;
+
+          case ActionConst.ModifyStackTypes.INSERT:
+            const index = command.beforeSceneKey ? findIndex(command.beforeSceneKey) : command.index;
```

Comment:
`line 272, col 1, Error - Line 272 exceeds the maximum line length of 100. (max-len)`